File: app/src/main/java/com/lomza/moviesroom/movie/MoviesViewModel.java
package com.lomza.moviesroom.movie;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;

import com.lomza.moviesroom.db.Movie;
import com.lomza.moviesroom.db.MovieDao;
import com.lomza.moviesroom.db.MoviesDatabase;

import java.util.List;

/**
 * @author Antonina
 */

public class MoviesViewModel extends AndroidViewModel {
    private MovieDao movieDao;
    private LiveData<List<Movie>> moviesLiveData;

    public MoviesViewModel(@NonNull Application application) {
        super(application);
        movieDao = MoviesDatabase.getDatabase(application).movieDao();
        moviesLiveData = movieDao.getAllMovies();
    }

    public LiveData<List<Movie>> getMoviesList() {
        return moviesLiveData;
    }

    public void insert(Movie... movies) {
        movieDao.insert(movies);
    }

    public void update(Movie movie) {
        movieDao.update(movie);
    }

    public void deleteAll() {
        movieDao.deleteAll();
    }
}

File: app/src/main/java/com/lomza/moviesroom/movie/MoviesViewModel.kt
package com.lomza.moviesroom.movie

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.lomza.moviesroom.db.*

/**
 * @author Antonina
 */
class MoviesViewModel(application: Application) : AndroidViewModel(application) {

    private val movieDao: MovieDao = MoviesDatabase.getDatabase(application).movieDao()
    private val directorDao: DirectorDao = MoviesDatabase.getDatabase(application).directorDao()

    val moviesList: LiveData<List<Movie>>
    val directorsList: LiveData<List<Director>>

    init {
        moviesList = movieDao.allMovies
        directorsList = directorDao.allDirectors
    }

    suspend fun insert(vararg movies: Movie) {
        movieDao.insert(*movies)
    }

    suspend fun update(movie: Movie) {
        movieDao.update(movie)
    }

    suspend fun deleteAll() {
        movieDao.deleteAll()
    }
}